### DISCUSSION TITLE
모니터링 메트릭, 로그 라벨 및 경보 정책 결정

### DISCUSSION THREAD

**Discussion by zeus6768** [woowacourse-teams/2024-code-zap#513] [결정 사항]
# 목적
- 특정 상황에 대한 알림
- 특정 상황이란?
  - [ ] (메트릭) 서버가 켜졌는지 꺼졌는지 알고 싶어요.
  - [ ] (메트릭)서버의 리소스 사용률을 알고 싶어요. 
  - [ ] (로그) 알 수 없는 오류(http status 5xx)가 발생할 때 알고 싶어요. 
  - [ ] (로그) 반복적인 Client(http status 4xx) 오류가 발생할 때 알고 싶어요. 
  - [ ] 악의적인 접근이 있을 경우 알고 싶어요. 
  - [ ] 누가 어떤 요청을 해서 어떤 결과가 나왔는지 파악하고 싶어요. 

# 안건

## 1. 메트릭 라벨
- 서버 실행 여부
- CPU, 메모리, 디스크 사용률
- 트래픽 in & out
  - 용량
  - 개수

## 2. 로그 라벨
- HTTP 요청 및 응답
  - 상태 코드
- 클라이언트 식별자

**Comment by kyum-q** [comment]
- [ ]  (메트릭) 서버가 켜졌는지 꺼졌는지 알고 싶어요.
- [ ]  (로그) 알 수 없는 오류(http status 5xx)가 발생할 때 알고 싶어요.

이전에 그라파나에서 알림을 구현한 적이 있는데 다음 포스트를 보고 확인했습니다 ~ 참고하면 좋을 것 같아요 ~

![image](https://github.com/user-attachments/assets/28ead20a-b675-426b-82d9-8692e58c6a61)


### 알림설정
로그: https://creampuffy.tistory.com/213
서버다운: https://yeees.tistory.com/459

### 알림 커스텀
https://jaeyung1001.tistory.com/entry/Grafana-Slack-Alert-%EB%A9%94%EC%84%B8%EC%A7%80-%EC%BB%A4%EC%8A%A4%ED%85%80%EB%A7%88%EC%9D%B4%EC%A7%95

**Comment by jminkkk** [comment]
- [ ]  error 레벨의 로그만 모아서 보고 싶어요
